type hints and comments for trtllm handler

ydm-amazon opened the pull request:
## Description ##

Trial commit for type hints and code clarity.

Only works for Python 3.9 and newer - older type hints style that is compatible with 3.8 is deprecated soon. Please let me know if compatibility with 3.8 is a must.

lanking520 commented:
python 3.9 is good enough. The only concern is inside Amazon there are still couple of teams using 3.8

lanking520 commented on engines/python/setup/djl_python/tensorrt_llm.py:
```diff
@@ -110,9 +141,15 @@ def inference(self, inputs):
 
 
-def handle(inputs: Input):
+def handle(inputs: Input) -> Optional[Output]:
```
why output is optional? We have to have Output object

ydm-amazon replied:
inputs, theoretically, could be None in an error case

```
    if inputs.is_empty():
        # initialization request
        return None
```

lanking520 replied:
None is a special treatment we deal at the back, I don't feel like making Optional is a good idea,

Does

```
Union[None, Output] = Optional[Output]
```
?

ydm-amazon replied:
Yes, the two are the same thing. Optional is just syntactic sugar

ydm-amazon replied:
It's fine to remove the Optional - there won't be an error thrown since I'm most likely doing type checking in another way that doesn't take information directly from type hints.

lanking520 replied:
@ydm-amazon one good thing of Union is that it support 3.8

ydm-amazon replied:
Oh, what I meant by not supporting 3.8 is choosing to not do

`from typing import List` and using `List` as a type hint

and instead just using `list` as a type hint. This is a feature introduced in 3.9, but the former way is either already deprecated or soon to be deprecated

ydm-amazon commented:
Added a separate commit so the two commits can be compared.

I changed the format of the comments to be the same as the currently existing format in other parts of djl-python. Previously, I was using pytorch format by default.
